public/index.js

```diff
@@ -1,23 +1,31 @@
 const user = {};
 
-const addChatMessageToDOM = (nickname, message) => {
+const addMessageToDOM = message => {
   const listItem = document.createElement('li');
-  listItem.textContent = `${nickname}: ${message}`;
+  if (message.type == 'chat') {
+    listItem.textContent = `${message.nickname}: ${message.body}`;
+  } else if (message.type == 'system') {
+    listItem.textContent = message.body;
+    listItem.style.fontStyle = 'italic';
+  }
   document.getElementById('messages').appendChild(listItem);
 };
 
 document.addEventListener('DOMContentLoaded', () => {
   const socket = io();
+  let message = {};
```

Comment:
Why did you initialize message up there? It's best to initialize variables as close to where they are used as possible.